scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,435 @@
+        )
+
+    output.status = status
+    output.report_json = report_json
+    output.alert = alert
+    output.message = message
+
+    print("Reading TXT report")
+    report_txt = "Not found"
+    if os.path.exists(TXT_REPORT_PATH):
+        with open(JSON_REPORT_PATH, mode="r") as handler:
+            report_txt = handler.read()
+
+    output.report = report_txt
+
+
+def update_insights_inventory():
+    """Call insights-client to update insights inventory."""
+    print("Updating system status in Red Hat Insights.")
+    output, returncode = run_subprocess(["/usr/bin/insights-client"])
+
+    if returncode:
+        raise ProcessError(
+            message="Failed to update Insights Inventory by registering the system again. See output the following output: %s"
+            % output,
+            report="insights-client execution exited with code '%s'." % returncode,
+        )
+
+    print("System registered with insights-client successfully.")
+
+
+def reboot_system():
+    print("Rebooting system in 1 minute.")
+    run_subprocess(["/usr/sbin/shutdown", "-r", "1"], wait=False)
+
+
+def main():
+    leapp_upgrade_output = None
+    try:
+        # Exit if not RHEL 7.9 or 8.4
+        dist, version = get_rhel_version()
+        if dist != "rhel" or is_non_eligible_releases(version):
+            raise ProcessError(
+                message="In-place upgrades are supported only on RHEL distributions.",
+                report='Exiting because distribution="%s" and version="%s"'
+                % (dist, version),
+            )
+
+        output = OutputCollector()
+        upgrade_command = ["/usr/bin/leapp", "upgrade", "--report-schema=1.1.0"]
+        rhui_pkgs = setup_leapp(version)
+
+        # Check for RHUI PKGs
+        use_no_rhsm = should_use_no_rhsm_check(len(rhui_pkgs) > 1, upgrade_command)
+        if use_no_rhsm:
+            install_leapp_pkg_corresponding_to_installed_rhui(rhui_pkgs)
+
+        remove_previous_reports()
+        leapp_upgrade_output = execute_upgrade(upgrade_command)
+        print("Leapp upgrade command successfully executed.")
+        parse_results(output)
+    except ProcessError as exception:
+        print(exception.report)
+        output = OutputCollector(
+            status="ERROR",
+            alert=True,
+            error=False,
+            message=exception.message,
+            report=exception.report,
+        )
+        leapp_upgrade_output = None
+    except Exception as exception:
+        print(str(exception))
+        output = OutputCollector(
+            status="ERROR",
+            alert=True,
+            error=False,
+            message="An unexpected error occurred. Expand the row for more details.",
+            report=str(exception),
+        )
+        leapp_upgrade_output = None
+    finally:
+        print("### JSON START ###")
+        print(json.dumps(output.to_dict(), indent=4))
+        print("### JSON END ###")
+        update_insights_inventory()
```

Comment:
I believe this should be called at the end of try block, only when the upgrade finishes, in finally it would be called even when error occurs and report is not generated. WDYT @r0x0d? In converions script it is same.

I changed it in https://github.com/oamg/leapp-insights-tasks/pull/4 as well.